jina/types/arrays/memmap.py

```diff
@@ -218,11 +233,13 @@ def __setitem__(self, key: Union[int, str], value: 'Document') -> None:
                 self.append(value)
                 self._header_map[self._int2str_id(key)] = self._header_map[value.id]
                 del self[value.id]
+                self._last_mmap = None
```

Comment:
maybe invalidations of `self._last_mmap` should happen inside each disk operation to the DAM, like append and delitem, not in methods that use them